src/main/kotlin/org/jetbrains/research/ml/codetracker/DocumentLogger.kt

```diff
@@ -1,92 +1,110 @@
 import com.intellij.openapi.application.PathManager
 import com.intellij.openapi.diagnostic.Logger
 import com.intellij.openapi.editor.Document
 import com.intellij.openapi.fileEditor.FileDocumentManager
 import com.intellij.openapi.util.io.FileUtil
-import data.DocumentChangeData
+import org.jetbrains.research.ml.codetracker.data.DocumentChangeData
 import org.apache.commons.csv.CSVFormat
 import org.apache.commons.csv.CSVPrinter
 import org.joda.time.DateTime
-import ui.ControllerManager
+import org.jetbrains.research.ml.codetracker.server.TrackerQueryExecutor
+import org.jetbrains.research.ml.codetracker.ui.ControllerManager
 import java.io.File
 import java.io.FileWriter
 import kotlin.math.abs
 
 
 object DocumentLogger {
-    private val diagnosticLogger: Logger = Logger.getInstance(javaClass)
+    data class Printer(val csvPrinter: CSVPrinter, val fileWriter: FileWriter, val file: File)
+
+    private val logger: Logger = Logger.getInstance(javaClass)
+
+    private val myDocumentsToPrinters: HashMap<Document, Printer> = HashMap()
 
-    val documentsToPrinters: HashMap<Document, Printer> = HashMap()
+    val documentsToPrinters: Map<Document, Printer>
+        get() = documentsToPrinters.toMap()
 
     private val folderPath = "${PathManager.getPluginsPath()}/code-tracker/"
     private const val MAX_FILE_SIZE = 50 * 1024 * 1024
     private const val MAX_DIF_SIZE = 300
 
-    data class Printer(val csvPrinter: CSVPrinter, val fileWriter: FileWriter, val file: File)
-
     fun log(document: Document) {
-        var printer = documentsToPrinters.getOrPut(document, { initPrinter(document) })
+        var printer = myDocumentsToPrinters.getOrPut(document, {
+            initPrinter(document)
+        })
         if (isFull(printer.file.length())) {
-            diagnosticLogger.info("${Plugin.PLUGIN_ID}: File ${printer.file.name} is full")
+            logger.info("${Plugin.PLUGIN_ID}: File ${printer.file.name} is full")
             sendFile(printer.file)
+            //TODO-birillo Should not you put here new printer to map?
             printer = initPrinter(document)
-            diagnosticLogger.info("${Plugin.PLUGIN_ID}: File ${printer.file.name} was cleared")
+            logger.info("${Plugin.PLUGIN_ID}: File ${printer.file.name} was cleared")
         }
         val change = document.getChange()
         printer.csvPrinter.printRecord(change.getData() + ControllerManager.uiData.getData().map { it.logValue })
     }
 
     fun logCurrentDocuments() {
-        diagnosticLogger.info("${Plugin.PLUGIN_ID}: log current documents: ${documentsToPrinters.keys.size}")
-        documentsToPrinters.keys.forEach { log(it) }
+        logger.info("${Plugin.PLUGIN_ID}: log current documents: ${myDocumentsToPrinters.keys.size}")
+        myDocumentsToPrinters.keys.forEach {
+            log(
+                it
+            )
+        }
     }
 
 
     private fun isFull(fileSize: Long): Boolean = abs(MAX_FILE_SIZE - fileSize) < MAX_DIF_SIZE
```

Comment:
Maybe we shouldn't limit the upper bound of fileSize? I mean, what if we get a fileSize bigger than MAX_FILE_SIZE + MAX_DIF_SIZE? That function considers that file as not full, but it definitely is full.